Modules/Devices/src/Devices.Application/Devices/Commands/DeleteDevice/Handler.cs

```diff
@@ -12,35 +13,27 @@ namespace Backbone.Modules.Devices.Application.Devices.Commands.DeleteDevice;
 public class Handler : IRequestHandler<DeleteDeviceCommand>
 {
     private readonly ChallengeValidator _challengeValidator;
-    private readonly IDevicesDbContext _dbContext;
     private readonly ILogger<Handler> _logger;
     private readonly IUserContext _userContext;
+    private readonly IDevicesRepository _devicesRepository;
 
-    public Handler(IDevicesDbContext dbContext, IUserContext userContext, ChallengeValidator challengeValidator, ILogger<Handler> logger)
+    public Handler(IDevicesRepository devicesRepository, IUserContext userContext, ChallengeValidator challengeValidator, ILogger<Handler> logger)
     {
-        _dbContext = dbContext;
+        _devicesRepository = devicesRepository;
         _userContext = userContext;
         _challengeValidator = challengeValidator;
         _logger = logger;
     }
 
     public async Task Handle(DeleteDeviceCommand request, CancellationToken cancellationToken)
     {
-        var device = await _dbContext.Set<Device>()
-            .OfIdentity(_userContext.GetAddress())
-            .NotDeleted()
-            .Include(d => d.Identity)
-            .FirstWithId(request.DeviceId, cancellationToken);
+        var device = await _devicesRepository.GetDeviceByIdentityAndId(_userContext.GetAddress(), request.DeviceId, cancellationToken);
```

Comment:
Get it only by Id and manually check if the device is part of the current Identity.

Reply:
May I ask why?

What should we do if the identity does not match?

Reply:
> May I ask why?
Of course. Because this is also application logic (Devices should only be fetchable by the owning Identity), and I don't want such logic in Infrastructure.

> What should we do if the identity does not match?
Throw a NotFoundException

Reply:
Makes sense, thanks